### PR TITLE
Fix packing/alignment issues in Win32 builds

### PR DESCRIPTION
**Pull request by afoxman**
The platformAdapters liblet had a cross-platform dependency on platform_posix, when the dependency should have been limited to posix-only platforms (everything but Windows). 

platform_posix includes empty pshpack8.h and poppack.h headers. On Windows builds, these were being used in place of the real headers, which actually change the packing size. To make things worse, they were mingled with pshpack2.h and pshpack4.h from the WinSDK. All of these packing headers are used by the Windows SDK (windows.h, winnt.h, ...).

This led to an unbalanced series of pragma pack push/pop operations which broke the whole header system.

While I was fixing this issue and verifying it in my own project, I ran into 2 C++17 bugs. std::async changed to [[nodiscard]] in C++17, causing build breaks in two UTs. I added code which effectively matches what you get in a C++14 build.

### CODE REVIEW

File: libs/object/tests/refCountedObjectTest.cpp [hunk 1]
```diff
@@ -430,7 +430,8 @@ struct AsyncDeleter
 		try
 		{
 			// Ideally we want to show here how to use dispatch queues, but we cannot add DispatchQueue liblet dependency here.
-			std::async(std::launch::async, [obj]() noexcept { TObject::RefCountPolicy::template Delete(obj); });
+			auto x = std::async(std::launch::async, [obj]() noexcept { TObject::RefCountPolicy::template Delete(obj); });
+			x.wait();
```
Comment (vmoroz) [inline libs/object/tests/refCountedObjectTest.cpp]: This causes the object to be destroyed synchronously that defeats the purpose of the unit test.
To avoid the compilation error, please add (void) in front of the std::async call instead:
(void)std::async(...)

The same is for the other change.

Comment (afoxman) [reply libs/object/tests/refCountedObjectTest.cpp]: I reverted to the original code, and instrumented the test to see what was actually happening. It wasn't working as you describe. std::async returns a std::future, which is immediately destroyed. The future dtor block until the async operation completes, as described [here](https://en.cppreference.com/w/cpp/thread/future/~future).

This is why I put in the wait call.

If you'd like I can change the test to use std::thread(functor).detach() which will let it run in the background and not block when the thread object destructs.

Comment (vmoroz) [reply libs/object/tests/refCountedObjectTest.cpp]: Ouch... Yes, please do the change to use the thread detach then. The goal of the test was to have the asynchronous destruction.

**Comment by afoxman** [comment]
I don't have write-access. Can you commit for me?
